README.md

```diff
@@ -147,6 +147,16 @@ WASI host must:
 A WASI host that implements the above should be able to spawn threads for a
 variety of languages.
 
+### TID (thread ID)
+
+TID is a 32-bit integer provided by the host to identify threads created
+with `wasi_thread_spawn`.
+
+* TID 0 is reserved. `wasi_thread_spawn` should not return this value.
+
+* The upper-most 3-bits of TID are always 0.
```

Comment:
note:

* the upper-most 2 bits are to meet the current assumptions in wasi_thread_spawn and wasi-libc.
* another bit is intended for https://github.com/WebAssembly/wasi-libc/pull/360 and similar usage.

Reply:
Also, I'm not sure if the proposal should be driven by the current implementation of it in other libraries; I think it should be other way around - unless it really makes sense to keep it like that.

Reply:
i think it really makes sense to avoid rewriting tid-based locking in musl.

Reply:
> Could you explain in the document why is that a requirement?

done